Vwitter/View Controllers/SignupViewController.m

```diff
@@ -0,0 +1,60 @@
+//
+//  SignupViewController.m
+//  Vwitter
+//
+//  Created by Christina Li on 7/8/22.
+//
+
+#import "SignupViewController.h"
+#import "VWUser.h"
+
+@interface SignupViewController ()
+@property (weak, nonatomic) IBOutlet UITextField *usernameField;
+@property (weak, nonatomic) IBOutlet UITextField *screennameField;
+@property (weak, nonatomic) IBOutlet UITextField *passwordField;
+
+@end
+
+@implementation SignupViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    // Do any additional setup after loading the view.
+}
+
+- (void)registerUser {
+
+    VWUser *newUser = [VWUser user];
+
+    newUser.username = self.usernameField.text;
+    newUser.password = self.passwordField.text;
+    newUser.screenName = self.screennameField.text;
```

Comment:
What happens if the user taps "Sign Up" button but didn't enter in any text for the password, or username, etc?

One of the most important things in programming is validating the data you are given by users, making sure it's not `nil`/empty, etc. At the very least, I would enforce a minimum character count for passwords. You can also easily scan the string to make sure it contains at least one capital letter, etc.

Reply:
Good point! I did some exploring and it seems that Parse's built-in user validation works pretty well; I tested it against a bunch of makeshift manual unit tests (ex. duplicate username, missing username, lack of password, etc.) and it seems to catch them all and give a specific error message that I pass on to the user :)

Reply:
For enforcing various password rules: should I do this on the Parse side or on my side? It seems that there's a Parse solution here: https://stackoverflow.com/questions/67030555/how-do-i-set-password-rules-when-signing-up-using-parse-and-swift

Reply:
Ultimately it is much better to enforce on the parse/server side, but in this case, it should be ok to just enforce a simple minimum length check in ObjC